Sources/Search/APIs/Custom.php

```diff
@@ -213,11 +213,10 @@ public function getSize(): int
 			if ($request !== false && Db::$db->num_rows($request) == 1) {
 				$row = Db::$db->fetch_assoc($request);
 				$this->size = (int) $row['Data_length'] + (int) $row['Index_length'];
+				Db::$db->free_result($request);
 			}
 		}
 
-		Db::$db->free_result($request);
-
```

Comment:
You'll also need to add a `Db::$db->free_result($request);` inside the PostgreSQL section of the code higher up.